internal/analytics/client/sync/client.go

```diff
@@ -49,7 +50,7 @@ type Client struct {
 var _ analytics.Dispatcher = &Client{}
 
 // New initializes the analytics instance with all custom dimensions known at this time
-func New(cfg *config.Instance, auth *authentication.Auth) *Client {
+func New(cfg *config.Instance, auth *authentication.Auth, out output.Outputer) *Client {
```

Comment:
The `out` argument is not being used anywhere in this PR, all instances of `sync.New()` is just passing `nil`

Reply:
For now the ticket is only interested in knowing if the state tool is running on CI and interactively. However, in developing the PR I thought we might eventually want to know if the installer or some other tool with synchronous analytics is being run on CI and interactively, so I created the plumbing for it without connecting anything.

Reply:
I think it makes sense, but why not actually use it? Instead of passing in nil, just pass in the outputer. I think we have it available in all the situations where you're currently passing in nil.